docs/node-operators/setup/generate-keys.md

```diff
@@ -1,58 +1,132 @@
 ---
-description: Securely generate private keys.
+description: Securely generate private keys for your validator node.
+title: Generate private keys
 ---
 
 import extension from './assets/extension.png';
 import mobile from './assets/mobile.png';
 
-# Generate keys
+Each validator node requires private keys for signing blocks and consensus messages sent to the other validators in the network.
 
-This guide demonstrates how to generate private keys for your nodes.
+The required private keys include one ECDSA key and one BLS key. The keys are named after digital signature schemes: Elliptic Curve Digital Signature Algorithm (ECDSA) and Boneh–Lynn–Shacham (BLS), respectively.
 
-## Generate locally
+- The ECDSA key is used for signing blocks.
+- The BLS key is used for producing finality votes. Validators who produce finality vote receive a reward. If a validator fails to produce a finality vote, no slashing penalty applies.
 
-### Prerequisites
+## Generate BLS keys
+
+You can generate a BLS key pair using the Ronin CLI or your node's Docker image.
+
+### Generate using the CLI
+
+1. Build Ronin from the source code:
+
+    ```
+    > git clone https://github.com/axieinfinity/ronin
+    > cd ronin
+    > make ronin
+    ```
+
+2. Add the Ronin binary path to the `$PATH` environment variable:
+
+    ```
+    > cp ./build/bin/ronin /usr/local/bin
+    ```
```

Comment:
The description is not correct. Technically, this copy the Ronin binary to /usr/local/bin and by default, the /usr/local/bin is already included in the $PATH variable. To add Ronin binary to $PATH, we need to append this line `PATH=$PATH:/path/to/ronin` into $HOME/.profile

```
Note: Changes made to a profile file may not apply until the next time you log into your computer. To apply the changes immediately, just run the shell commands directly or execute them from the profile using a command such as source $HOME/.profile.
```
Copied from https://go.dev/doc/install